platforms/ide/problems-api/src/main/java/org/gradle/api/problems/internal/DefaultBuildableProblemBuilder.java

```diff
@@ -147,35 +150,19 @@ private ReportableProblem buildInternal(@Nullable Severity severity) {
             throw new IllegalStateException("Problem is not documented: " + label);
         }
 
-        if (!noLocation) {
-            if (path == null) {
-                throw new IllegalStateException("Problem location path is not set: " + label);
-            }
-            if (line == null) {
-                throw new IllegalStateException("Problem location line is not set: " + label);
-            }
-            // Column is optional field, so we don't need to check it
-        }
-
         return new DefaultReportableProblem(
             label,
             getSeverity(severity),
-            getProblemLocation(),
+            locations,
             documentationUrl,
             description,
             solution,
-            exception,
+            exception == null ? new Exception() : exception,
```

Comment:
I will be frank, I forgot why we added this.
@donat do you remember anything?

Reply:
Here's what I remember: we can get location data from a stacktrace of an exception. If the error report does not provide an exception `new Exception()` would be a fallback for analysis.

Reply:
You are right, it would be used to feed [StackLocationTransformer.java](https://github.com/gradle/gradle/pull/26548/files#diff-6796748a9a1942aa0367d4965cf898088e75a181426e84d4549e19f46cb0a7c5) data.